releng/docker/Dockerfile

```diff
@@ -7,34 +9,99 @@
+# default to build ZFSBootMenu images (default: master)
 ARG ZBM_COMMIT_HASH
+# Include the specified Void Linux package in the image
+#
+# (multiple entries must be seperated by spaces)
+ARG PACKAGES=
+ARG XBPS_REPOS="https://repo-fastly.voidlinux.org/current"
+# Include the specified Void Linux kernel series in the image; kernel
+# series take the form <major>.<minor> and correspond to Void packages
+# linux<kver> and linux<kver>-headers
+#
+# Default: install 5.10, 5.15 and 6.1
+#	
+# (multiple entries must be seperated by spaces)
+ARG KERNELS="linux5.10 linux5.15 linux6.1" 
+
+RUN mkdir -p /etc/xbps.d
+RUN <<EOF
+for repo in "${XBPS_REPOS}"; do
+  echo "repository=$repo" >> /etc/xbps.d/00-custom-repos.conf
+done
+EOF
 
 # Ensure everything is up-to-date
-RUN xbps-install -Suy xbps && xbps-install -uy
+RUN xbps-install -Suy xbps && xbps-install -uy bash
+SHELL ["/bin/bash", "-c"]
+
+# Install custom packages
+RUN <<EOF
+if [ -n "${PACKAGES}" ]; then
+  	echo "INFO: adding extra Void packages: ${PACKAGES}"
+	xbps-install -y $PACKAGES
+fi
+EOF
 
 # Prefer an LTS version over whatever Void thinks is current
-RUN echo "ignorepkg=linux" > /etc/xbps.d/10-nolinux.conf \
-	&& echo "ignorepkg=linux-headers" >> /etc/xbps.d/10-nolinux.conf
+RUN <<EOF
+echo "ignorepkg=linux" >> /etc/xbps.d/10-nolinux.conf
+echo "ignorepkg=linux-headers" >> /etc/xbps.d/10-nolinux.conf
+EOF
 
-# Install components necessary to build the image
+# Prevent initramfs kernel hooks from being installed
+RUN <<EOF
+echo "noextract=/usr/libexec/mkinitcpio/*" >> /etc/xbps.d/15-noinitramfs.conf
+echo "noextract=/usr/libexec/dracut/*" >> /etc/xbps.d/15-noinitramfs.conf
+EOF
+
+# Install ZFSBootMenu dependencies and components necessary to build images
 RUN xbps-query -Rp run_depends zfsbootmenu | xargs xbps-install -y
-RUN xbps-install -y linux5.10 linux5.10-headers \
-	zfs gummiboot-efistub curl yq-go bash kbd \
-	terminus-font dracut mkinitcpio cryptsetup
+RUN <<EOF
+if [ -n "${KERNELS}" ]; then
+	for _kern in "${KERNELS}"; do
+		kern_headers+=( "${_kern}-headers" )
+	done
+	echo Installing Kernels: ${KERNELS} ${kern_headers}
+	xbps-install -y ${KERNELS} ${kern_headers}
+fi
+EOF
 
-# Remove headers and massive dkms development toolchain; binutils
-# provides objcopy, which is necessary for UEFI bundle creation
-RUN xbps-pkgdb -m manual binutils
+RUN xbps-install -y zstd gummiboot-efistub curl yq-go bash kbd terminus-font \
+  dracut mkinitcpio dracut-network gptfdisk iproute2 iputils parted curl \
+  dosfstools e2fsprogs efibootmgr cryptsetup openssh util-linux kpartx
+
+# Remove headers and development toolchain, but keep binutils for objcopy
 RUN echo "ignorepkg=dkms" > /etc/xbps.d/10-nodkms.conf
-RUN xbps-remove -Roy linux5.10-headers dkms && rm -f /var/cache/xbps/*
+RUN xbps-pkgdb -m manual binutils
+
+RUN <<EOF
+if [ -n "${KERNELS}" ]; then
+	for _kern in "${KERNELS}"; do
+		kern_headers+=( "${_kern}-headers" )
+	done
+	xbps-remove -Roy ${kern_headers}
+fi
```

Comment:
Same as above: `$KERNELS` probably shouldn't be quoted in the loop, and testing that it's non-empty doesn't make sense.